### PR TITLE
feat: vexhub support

### PR DESCRIPTION
**Pull request by alegrey91**
## Description

Fix #XXX

## Test

To test this pull request, you can run the following commands:

### CODE REVIEW

File: internal/handlers/scan_sbom.go [hunk 1]
```diff
@@ -119,8 +134,47 @@ func (h *ScanSBOMHandler) Handle(ctx context.Context, message []byte) error { //
...
+	}
+
+	if len(vexHubList.Items) > 0 {
+		trivyVEXPath := path.Join(trivyHome, trivyVEXSubPath)
+		vexRepoPath := path.Join(trivyVEXPath, trivyVEXRepoFile)
+		if err = h.setupVEXHubRepositories(vexHubList, trivyVEXPath, vexRepoPath); err != nil {
+			return fmt.Errorf("failed to setup VEX Hub repositories: %w", err)
+		}
+		// Clean up the repository.yaml file after each handler execution to
+		// ensure VEX repositories are refreshed on every run.
```
Comment (fabriziosestito) [inline internal/handlers/scan_sbom.go:159]: This comment is outdated with the latest changes. Here, we are cleaning up the entire temp directory used as the Trivy home.

File: internal/handlers/scan_sbom_test.go [hunk 2]
```diff
@@ -22,54 +26,100 @@ import (
...
+						Enabled: true,
+					},
+				},
+			},
+			sourceSBOMJSON:     filepath.Join("..", "..", "test", "fixtures", "golang-1.12-alpine-s390x.spdx.json"),
+			expectedReportJSON: filepath.Join("..", "..", "test", "fixtures", "golang-1.12-alpine-s390x.sarif.json"),
+		},
+		{
+			platform: "linux/s390x with VEX repo not enabled",
+			vexHubList: []v1alpha1.VEXHub{
```
Comment (fabriziosestito) [inline internal/handlers/scan_sbom_test.go:101]: Could we simplify this by adding a vexHub variable directly to the test case, set to nil when there's no VEXHub, and assigned a value when we want to test with on, instead of using a full list?

**Review by fabriziosestito — CHANGES_REQUESTED** [review]
Overall, great work!
A few small changes have been requested.

File: internal/handlers/scan_sbom.go [hunk 3]
```diff
@@ -165,3 +219,40 @@ func (h *ScanSBOMHandler) Handle(ctx context.Context, message []byte) error { //
...
+	}
+
+	var repositories []byte
+	repositories, err = yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal struct: %w", err)
+	}
+
+	h.logger.Debug("Creating VEX repository directory", "vexhub", trivyVEXPath)
+	if _, err = os.Stat(trivyVEXPath); os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (alegrey91) [reply internal/handlers/scan_sbom.go:244]: nice, I didn't know about that. thanks for explaining.

File: internal/handlers/scan_sbom_test.go (hunk 2 above)
Comment (fabriziosestito) [reply internal/handlers/scan_sbom_test.go:101]: never mind, we agreed to leave it like it is for simplicity

**Review by fabriziosestito — APPROVED** [review]

File: internal/handlers/scan_sbom.go [hunk 4]
```diff
@@ -7,22 +7,32 @@ import (
 	"io"
 	"log/slog"
 	"os"
+	"path"
 
+	"gopkg.in/yaml.v3"
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply internal/handlers/scan_sbom.go:12]: @flavio @alegrey91 we already import [sigs.k8s.io/yaml](https://pkg.go.dev/sigs.k8s.io/yaml) as indirect, could we use this one instead?

[the remark this answers is not in the document]
Comment (olblak) [reply internal/handlers/scan_sbom.go]: @fabriziosestito The kubernetes project is consolidating on github.com/yaml/go-yaml which is one the dependency of sigs.k8s.io/yaml

cfr https://github.com/kubernetes/kubernetes/issues/132056

https://github.com/kubernetes-sigs/yaml/blob/048d724aca2d37ddb5b03c90b5b4550a3a48766d/go.mod#L8C2-L8C20

IMHO it's better use "github.com/yaml/go-yaml" directly

File: internal/handlers/scan_sbom_test.go [hunk 5]
```diff
@@ -22,54 +26,100 @@ import (
...
+						Name: "test",
+					},
+					Spec: v1alpha1.VEXHubSpec{
+						URL:     vexHubServer.URL,
+						Enabled: true,
+					},
+				},
+			},
+			sourceSBOMJSON:     filepath.Join("..", "..", "test", "fixtures", "golang-1.12-alpine-s390x.spdx.json"),
+			expectedReportJSON: filepath.Join("..", "..", "test", "fixtures", "golang-1.12-alpine-s390x.sarif.json"),
```
[the remark this answers is not in the document]
Comment (flavio) [reply internal/handlers/scan_sbom_test.go:97]: Fine with me. @alegrey91, please file an issue to not forget about that

File: internal/handlers/scan_sbom.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (flavio) [reply internal/handlers/scan_sbom.go]: @alegrey91 sorry to bother you again, can you switch to `go.yaml.in/yaml/v3 v3.0.3`? Thanks

File: internal/handlers/scan_sbom_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (alegrey91) [reply internal/handlers/scan_sbom_test.go:97]: here it is: https://github.com/rancher-sandbox/sbombastic/issues/341

**Review by flavio — APPROVED** [review]
